File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/MvpDelegateCallback.java
package me.ameriod.lib.mvp.view.deligate;

import android.support.annotation.NonNull;

import me.ameriod.lib.mvp.Mvp;

public interface MvpDelegateCallback<V extends Mvp.View, P extends Mvp.Presenter<V>> {

    @NonNull
    P createPresenter();

    P getPresenter();

    V getMvpView();
}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/MvpDelegateCallback.kt
package me.ameriod.lib.mvp.deligate

import me.ameriod.lib.mvp.Mvp

interface MvpDelegateCallback<out V : Mvp.View, out P : Mvp.Presenter<out V>> {

    fun createPresenter(): P

    fun getPresenter(): P

    fun getMvpView(): V
}